WordPress/Classes/Models/Blog.m

```diff
@@ -439,6 +444,46 @@ - (BOOL)supports:(BlogFeature)feature
     }
 }
 
+- (BOOL)supportsPublicize
+{
+    // wpcom blog and publicize has not been disabled
+    if (self.isHostedAtWPcom && ![[self getOptionValue:OptionsKeyPublicizeDisabled] boolValue]) {
+        return YES;
+    }
+
+    // self-hosted connected to jetpack via the wpcom acct signed into the app.
+    if (self.dotComID && [self jetpackPublicizeModuleEnabled]) {
+        return YES;
+    }
+
+    // self-hosted connected to jetpack via a jetpackAccount
+    if ([self jetpackRESTSupported] && [self jetpackPublicizeModuleEnabled]) {
+        return YES;
+    }
+
+    return NO;
+}
+
+- (BOOL)supportsShareButtons
+{
+    // wpcom blog
+    if (self.isHostedAtWPcom) {
+        return YES;
+    }
+
+    // self-hosted connected to jetpack via the wpcom acct signed into the app.
+    if (self.dotComID && [self jetpackSharingButtonsModuleEnabled]) {
+        return YES;
+    }
```

Comment:
Fwiw, I considered consolidating the two if blocks into something like 
`if ((self.dotComID || [self jetpackRESTSupported]) && [self jetpackSharingButtonsModleEnabled)` but I decided separating the checks was better for clarity of purpose.